modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,290 @@
+ */
+function webp_uploads_get_supported_image_mime_transforms() {
+	$image_mime_transforms = array(
+		'image/jpeg' => array( 'image/webp' ),
+		'image/webp' => array( 'image/jpeg' ),
+	);
+
+	/**
+	 * Filter to allow the definition of a custom mime types, in which a defined mime type
+	 * can be transformed and provide a wide range of mime types.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, array<string>> An array with the valid mime transforms.
+	 *
+	 * @return array<string, array<string>> An array with the valid mime transformation
+	 */
+	return (array) apply_filters( 'webp_uploads_supported_image_mime_transforms', $image_mime_transforms );
+}
+
+/**
+ * Hook fired when an attachment is deleted, this hook is in charge of removing any
+ * additional mime types created by this plugin besides the original image. Any source
+ * with the same as the main image would not be removed by this hook due this file would
+ * be removed by WordPress when the attachment is deleted, usually this happens after this
+ * hook is executed.
+ *
+ * @since n.e.x.t
+ *
+ * @see wp_delete_attachment
+ *
+ * @param int $attachment_id The ID of the attachment the sources are going to be deleted.
+ */
+function webp_uploads_remove_sources_files( $attachment_id ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+	$file     = get_attached_file( $attachment_id );
+
+	if (
+		! isset( $metadata['sizes'] )
+		|| empty( $file )
+		|| ! is_array( $metadata['sizes'] )
+	) {
+		return;
+	}
+
+	$upload_path = wp_get_upload_dir();
+	if ( empty( $upload_path['basedir'] ) ) {
+		return;
+	}
+
+	$intermediate_dir = path_join( $upload_path['basedir'], dirname( $file ) );
+	$basename         = wp_basename( $file );
+
+	foreach ( $metadata['sizes'] as $size ) {
+		if ( ! isset( $size['sources'] ) || ! is_array( $size['sources'] ) ) {
+			continue;
+		}
+
+		$original_size_mime = empty( $size['mime-type'] ) ? '' : $size['mime-type'];
+
+		foreach ( $size['sources'] as $mime => $properties ) {
+			/**
+			 * When we face the same mime type as the original image, we ignore this file as this file
+			 * would be removed when the size is removed by WordPress itself. The meta information as well
+			 * would be deleted as soon as the image is removed.
+			 *
+			 * @see wp_delete_attachment
+			 */
+			if ( $original_size_mime === $mime ) {
+				continue;
+			}
+
+			if ( ! is_array( $properties ) || empty( $properties['file'] ) ) {
+				continue;
+			}
+
+			$intermediate_file = str_replace( $basename, $properties['file'], $file );
+			if ( ! empty( $intermediate_file ) ) {
+				$intermediate_file = path_join( $upload_path['basedir'], $intermediate_file );
+				wp_delete_file_from_directory( $intermediate_file, $intermediate_dir );
+			}
+		}
+	}
 }
-add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
```

Comment:
leave this for when `webp_uploads_get_supported_image_mime_transforms()` returns only webp.

Reply:
Well the transform won't define which mime types are allowed instead allows to define in which mime types the provided mime type can be transformed, if the mime is not present on the array no additional mime type would be created for that image but the image would be uploaded as it was uploaded with no additional change.

Reply:
See my response in https://github.com/WordPress/performance/pull/147#discussion_r813265136 - right now this is without a purpose because the new default behavior is different. We should reintroduce something like this again once we have actually defined the API for developers to tweak the behavior. We can then use this function as a starting point to tweak.

Reply:
👍🏼